### ISSUE TITLE
Parser infinite loop when seeing unsupported operator in set while parsing return

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/jakt#218]
This causes an infinite loop in the parser:
```jakt
function foo() {
    return {1:1}
}
```
Removing `return` makes it error out correctly:
```
-----
function foo() {
    {1:1}
}
 
-----
Error: ParserError("unsupported expression", Span { file_id: 1, start: 23, end: 24 
```

**Comment by holychowders** [comment]
Looking into this.

**Comment by lanmonster** [comment]
fixed as of 9012330
